pkg/services/ngalert/api/tooling/definitions/prom.go

```diff
@@ -168,12 +169,10 @@ type AlertingRule struct {
 // adapted from cortex
 // swagger:model
 type Rule struct {
+	UID string `json:"uid,omitempty"`
 	// required: true
-	UID string `json:"uid"`
-	// required: true
-	Name string `json:"name"`
-	// required: true
-	FolderUID string `json:"folderUid"`
+	Name      string `json:"name"`
+	FolderUID string `json:"folderUid,omitempty"`
```

Comment:
was the removal of the `required: true` intentional?

Reply:
Yes. Unfortunatelly, the `Rule` struct seems to be shared between GMA and DMA rules, and with `required: true` the API returns empty string for DMA rules for `FolderUID` and `UID` properties
If there is any better way to fix this I'm happy to do that